sw/airborne/modules/computer_vision/opticflow/opticflow_calculator.c

```diff
@@ -308,6 +328,167 @@ void opticflow_calc_frame(struct opticflow_t *opticflow, struct opticflow_state_
+   * */
+  edgeflow.flow_x = -1 * edgeflow.flow_x;
+  edgeflow.flow_y = -1 * edgeflow.flow_y;
+
+  result->flow_x = (int16_t)edgeflow.flow_x / previous_frame_offset[0];
+  result->flow_y = (int16_t)edgeflow.flow_y / previous_frame_offset[1];
+
+  //Fill up the results optic flow to be on par with LK_fast9
+  result->flow_der_x =  result->flow_x;
+  result->flow_der_y =  result->flow_y;
+  result->corner_cnt = getAmountPeaks(edge_hist_x, 500 , img->w);
+  result->tracked_cnt = getAmountPeaks(edge_hist_x, 500 , img->w);
+  result->divergence = (float)edgeflow.flow_x / RES;
+  result->div_size = 0.0f;
+  result->noise_measurement = 0.0f;
+  result->surface_roughness = 0.0f;
+
+  //......................Calculating VELOCITY ..................... //
+
+  /*Estimate fps per direction
+   * This is the fps with adaptive horizon for subpixel flow, which is not similar
+   * to the loop speed of the algorithm. The faster the quadcopter flies
+   * the higher it becomes
+  */
+  float fps_x = 0;
+  float fps_y = 0;
+  float time_diff_x = (float)(timeval_diff(&edge_hist[previous_frame_nr[0]].frame_time, &img->ts)) / 1000.;
+  float time_diff_y = (float)(timeval_diff(&edge_hist[previous_frame_nr[1]].frame_time, &img->ts)) / 1000.;
+  fps_x = 1 / (time_diff_x);
+  fps_y = 1 / (time_diff_y);
+
+  result->fps = fps_x;
+
+  // Calculate velocity
+  float vel_x = edgeflow.flow_x * fps_x * state->agl * OPTICFLOW_FOV_W / (img->w * RES);
+  float vel_y = edgeflow.flow_y * fps_y * state->agl * OPTICFLOW_FOV_H / (img->h * RES);
+  result->vel_x = vel_x;
+  result->vel_y = vel_y;
+
+  /* Rotate velocities from camera frame coordinates to body coordinates.
+  * IMPORTANT This frame to body orientation should bethe case for the parrot
+  * ARdrone and Bebop, however this can be different for other quadcopters
+  * ALWAYS double check!
+  */
+  result->vel_body_x = - vel_y;
+  result->vel_body_y = vel_x;
+
+#if OPTICFLOW_DEBUG && OPTICFLOW_SHOW_FLOW
+  draw_edgeflow_img(img, edgeflow, displacement, *edge_hist_x)
+#endif
+  // Increment and wrap current time frame
+  current_frame_nr = (current_frame_nr + 1) % MAX_HORIZON;
+}
+
+
+/**
+ * Run the optical flow on a new image frame
+ * @param[in] *opticflow The opticalflow structure that keeps track of previous images
+ * @param[in] *state The state of the drone
+ * @param[in] *img The image frame to calculate the optical flow from
+ * @param[out] *result The optical flow result
+ */
+void opticflow_calc_frame(struct opticflow_t *opticflow, struct opticflow_state_t *state, struct image_t *img,
+                          struct opticflow_result_t *result)
+{
+  if (opticflow->method == 0) {
+    calc_fast9_lukas_kanade(opticflow, state, img, result);
+  } else {
+    if (opticflow->method == 1) {
+      calc_edgeflow_tot(opticflow, state, img, result);
+    } else {
+      PRINT_CONFIG_MSG("Both edgeflow and Lukas kanade is not turned on. Define either USE_LK or use_EDGEFLOW on TRUE!");
```

Comment:
This does not work as you probably intended...
`PRINT_CONFIG_MSG` will print the message during compilation if the "print config" checkbox is enabled.
Since this already a preprocessor macro, the else condition does nothing here and it will always be printed.